### PR TITLE
add parameters to with_launch_file decorator #29

### PR DESCRIPTION
**Pull request by schmidta83**
Implements #29.

### CODE REVIEW

File: ros2_easy_test/decorators.py [hunk 1]
```diff
@@ -215,6 +216,7 @@ def with_launch_file(  # noqa: C901
         launch_file: Either:
             1) The path to the launch file to start for the test.
             2) The literal launch file (must contain a newline to be detected as such).
+        parameters: The parameters to be set for the node as ``("key", value)`` pairs
```
Comment (felixdivo) [inline ros2_easy_test/decorators.py:219]: Here, the key is escaped, but below it is not. Should we escape them on the command line?

**Review by felixdivo — CHANGES_REQUESTED** [review]
Nice contribution!

We should also add a small test for it, though. Could be similar to this: https://github.com/felixdivo/ros2-easy-test/blob/47590ce72a23f4f939d1e1f06640c9102c531617/tests/test_interactions.py#L106-L109

File: ros2_easy_test/decorators.py (hunk 1 above)
Comment (schmidta83) [reply ros2_easy_test/decorators.py:219]: I do not get exactly what you mean with "below". Here it is just the description of the function argument.

File: ros2_easy_test/decorators.py [hunk 2]
```diff
@@ -253,9 +255,13 @@ def wrapper(*args_inner, **kwargs_inner) -> None:
                     "ros2",
                     "launch",
                     str(launch_file_path),
-                    "--noninteractive",
```
[the remark this answers is not in the document]
Comment (schmidta83) [reply ros2_easy_test/decorators.py:256]: Jes it does, otherwise the node can not be launched properly

**Comment by Timple** [comment]
Wouldn't the term `launch_arguments` be more correct? You cannot add parameters to a launchfile. As it wouldn't know to which node to add the parameters.

File: ros2_easy_test/decorators.py (hunk 1 above)
Comment (felixdivo) [reply ros2_easy_test/decorators.py:219]: I meant in the string that is built below, there are not `"`s. But the documentation suggests there are.

**Comment by felixdivo** [comment]
> Wouldn't the term launch_arguments be more correct? You cannot add parameters to a launchfile. As it wouldn't know to which node to add the parameters.

I think you are right. Thanks for checking this!

@schmidta83 We can keep all as-is; just need to rename the parameter and docstring above to call it "launch arguments passed to the launch file" instead of "parameters passed to the node". :)

File: ros2_easy_test/decorators.py (hunk 1 above)
Comment (felixdivo) [reply ros2_easy_test/decorators.py:219]: Okay, I guess it does not really make sense to support keys or values that need escaping here. And if a user really needs it, they can still escape it themselves.

Let's just do this (see below for the discussion):

 ```suggestion
         arguments: The arguments for the launch file, which will be passed as ``key:=value`` pairs
```

**Comment by schmidta83** [comment]
Sorry, was busy last weeks...
@Timple , good point! Will change it.

@felixdivo Thanks for the information.

**Comment by felixdivo** [comment]
No worries, @schmidta83. I was unresponsive for weeks, too. 🙃

**Comment by schmidta83** [comment]
@felixdivo All suggegtions are adapted. You can review again.

**Review by felixdivo — APPROVED** [review]
Nice!

**Pull request merged**